blobmodel/model.py

```diff
@@ -97,42 +106,68 @@ def make_realization(
             t_drain=self.t_drain,
         )
 
-        output = np.zeros(
+        __density = np.zeros(
             shape=(self.__geometry.Ny, self.__geometry.Nx, self.__geometry.t.size)
         )
+        if labels:
+            __labels_field = np.zeros(
+                shape=(self.__geometry.Ny, self.__geometry.Nx, self.__geometry.t.size)
+            )
 
         for b in tqdm(self.__blobs, desc="Summing up Blobs"):
             # speedup implemeted for exponential pulses
             # can also be used for gaussian pulses since they converge faster than exponential pulses
             if speed_up:
-                start = int(b.t_init / self.__geometry.dt)
+                __start = int(b.t_init / self.__geometry.dt)
                 if b.v_x == 0:
-                    stop = self.__geometry.t.size
+                    __stop = self.__geometry.t.size
                 else:
                     # ignores t_drain when calculating stop time
-                    stop = start + int(
-                        (-np.log(error * np.sqrt(np.pi)) + self.__geometry.Lx - b.pos_x)
-                        / (b.v_x * self.__geometry.dt)
+                    __stop = np.minimum(
+                        self.__geometry.t.size,
+                        __start
+                        + int(
+                            (
+                                -np.log(error * np.sqrt(np.pi))
+                                + self.__geometry.Lx
+                                - b.pos_x
+                            )
+                            / (b.v_x * self.__geometry.dt)
+                        ),
                     )
-                output[:, :, start:stop] += b.discretize_blob(
-                    x=self.__geometry.x_matrix[:, :, start:stop],
-                    y=self.__geometry.y_matrix[:, :, start:stop],
-                    t=self.__geometry.t_matrix[:, :, start:stop],
+                __single_blob = b.discretize_blob(
+                    x=self.__geometry.x_matrix[:, :, __start:__stop],
+                    y=self.__geometry.y_matrix[:, :, __start:__stop],
+                    t=self.__geometry.t_matrix[:, :, __start:__stop],
                     periodic_y=self.__geometry.periodic_y,
                     Ly=self.__geometry.Ly,
                 )
+                __density[:, :, __start:__stop] += __single_blob
+                if labels:
+                    __max_amplitudes = np.max(__single_blob, axis=(0, 1))
+                    __max_amplitudes[__max_amplitudes == 0] = np.inf
+                    __tmp = np.copy(__labels_field[:, :, __start:__stop])
+                    __tmp[__single_blob >= __max_amplitudes * label_border] = 1
+                    __labels_field[:, :, __start:__stop] = __tmp
```

Comment:
Why create __tmp? Seems unnecessary: 

__labels_field[:, :, __start:__stop][__single_blob >= __max_amplitudes * label_border] = 1

Reply:
I couldn't find another way of implementing this since I am not aware of a simpler way of applying a condition only on a slice of the array.
i.e.
__label_field[:, :, __start:__stop &&  __single_blob >= __max_amplitudes * label_border]

@Sosnowsky do you know a simpler way in python? Otherwise I would keep the current implementation

Reply:
Doesn't what I wrote work?
__labels_field[:, :, __start:__stop][__single_blob >= __max_amplitudes * label_border] = 1

Creating the tmp array is time-comsuming, so in general you would prefer to avoid doing it if it is not necessary.